internal/form/handler.go

```diff
@@ -0,0 +1,310 @@
+package form
+
+import (
+	"NYCU-SDC/core-system-backend/internal/form/question"
+	"NYCU-SDC/core-system-backend/internal/user"
+	"context"
+	"fmt"
+	handlerutil "github.com/NYCU-SDC/summer/pkg/handler"
+	logutil "github.com/NYCU-SDC/summer/pkg/log"
+	"github.com/NYCU-SDC/summer/pkg/problem"
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgtype"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+	"net/http"
+)
+
+type FormRequest struct {
+	Title       string `json:"title" validate:"required"`
+	Description string `json:"description"`
+}
+
+type QuestionRequest struct {
+	FormID      uuid.UUID
+	Required    bool   `json:"required" validate:"required"`
+	Type        string `json:"type" validate:"required,oneof=short_text long_text single_choice multiple_choice date"`
+	Label       string `json:"label" validate:"required"`
+	Description string `json:"description"`
+	Order       int32  `json:"order" validate:"required"`
+}
+
+type FormStore interface {
+	Create(ctx context.Context, request FormRequest, userid uuid.UUID) (Form, error)
+	Update(ctx context.Context, id uuid.UUID, request FormRequest, userid uuid.UUID) (Form, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+	GetByID(ctx context.Context, id uuid.UUID) (Form, error)
+	List(ctx context.Context) ([]Form, error)
+}
+
+type QuestionStore interface {
+	Create(ctx context.Context, input question.CreateParams) (question.Question, error)
+	Update(ctx context.Context, input question.UpdateParams) (question.Question, error)
+	Delete(ctx context.Context, formID uuid.UUID, id uuid.UUID) error
+	ListByFormID(ctx context.Context, formID uuid.UUID) ([]question.Question, error)
+}
+
+type Handler struct {
+	logger *zap.Logger
+	tracer trace.Tracer
+
+	validator     *validator.Validate
+	problemWriter *problem.HttpWriter
+
+	formStore     FormStore
+	questionStore QuestionStore
+}
+
+func NewHandler(
+	logger *zap.Logger,
+	validator *validator.Validate,
+	problemWriter *problem.HttpWriter,
+	formStore FormStore,
+	questionStore QuestionStore,
+) *Handler {
+	return &Handler{
+		logger:        logger,
+		tracer:        otel.Tracer("form/handler"),
+		validator:     validator,
+		problemWriter: problemWriter,
+		formStore:     formStore,
+		questionStore: questionStore,
+	}
+}
+
+func (h *Handler) CreateFormHandler(w http.ResponseWriter, r *http.Request) {
+	traceCtx, span := h.tracer.Start(r.Context(), "CreateFormHandler")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, h.logger)
+
+	var req FormRequest
+	if err := handlerutil.ParseAndValidateRequestBody(traceCtx, h.validator, r, &req); err != nil {
+		h.problemWriter.WriteError(traceCtx, w, err, logger)
+		return
+	}
+
+	user, ok := user.GetUserFromContext(traceCtx)
```

Comment:
Consider using `currentUser` instead of `user`, since user is reserved for the package name and may pollute the namespace.